### PR TITLE
10 post/article id/comments

### PR DESCRIPTION
**Pull request by MrMcSnail**
Adds a new endpoint which posts a new comment based on the article_id. 
It accepts an object with a username property which only accepts an existing username and a body property.

error in case the article_id does not exist within the data base

error when given an invalid format within the request body.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -185,3 +185,46 @@ describe('GET /api/articles/:article_id/comments', () => {
...
+      expect(body.comment).toEqual(expect.objectContaining(commentObj));
+    });
+  });
+  test('Status 404: should respond with ""article_id" is an invalid Article ID." when an incorrect article id is passed', ()=>{
+    return request(app)
+        .post('/api/articles/10101/comments')
+        .send({username: 'rogersop', body: `this article doesn't even exist`})
+        .expect(404)
+        .then(({body})=>{
+          expect(body.msg).toBe(`10101 is an invalid Article ID.`);
```
Comment (cjplatten) [inline __tests__/app.test.js:209]: As a number, is 10101 an invalid ID?

File: __tests__/app.test.js [hunk 2]
```diff
@@ -185,3 +185,46 @@ describe('GET /api/articles/:article_id/comments', () => {
...
+        });
+  });
```
Comment (cjplatten) [inline __tests__/app.test.js:211]: What response would you get from the the server if the ID was not a number?

File: __tests__/app.test.js [hunk 3]
```diff
@@ -185,3 +185,46 @@ describe('GET /api/articles/:article_id/comments', () => {
...
+  test('Status 400: should respond with "Invalid Input" when given an invalid format within the request body', ()=>{
+    return request(app)
+        .post('/api/articles/1/comments')
+        .send({username: 'rogersop', handbag: 123})
+        .expect(400)
+        .then(({body})=>{
+          expect(body.msg).toBe(`Invalid Input`);
+        });
+  });
+  test('Status 400: responds with "Username Does Not Exist" when the request includes a username not in the database', ()=>{
```
Comment (cjplatten) [inline __tests__/app.test.js:221]: what status is used when something can't be found?

**Review by cjplatten — CHANGES_REQUESTED** [review]
Please see the comments, this is a good start!

File: __tests__/app.test.js (hunk 2 above)
Comment (MrMcSnail) [reply __tests__/app.test.js:211]: Status 400: should respond with Invalid Input when article_id is of the wrong type

File: __tests__/app.test.js (hunk 3 above)
Comment (MrMcSnail) [reply __tests__/app.test.js:221]: 404! 🤦

**Pull request merged**